### ISSUE TITLE
CLI app

### ISSUE THREAD

**Issue by fabiospampinato** [notable/notable#667]
### Feature description

An official CLI app would be useful.

### Feature motivation

It would have multiple purposes: starting the app with some flags/options, it could provide third-party tools some features like searching etc.

**Comment by divyanshu013** [comment]
Regarding search, it could be perhaps something like `fzf` has to quickly search and open notes in the app.

**Comment by fabiospampinato** [comment]
@divyanshu013 Interface-wise?

**Comment by divyanshu013** [comment]
Yes, perhaps `fzf` can be used directly (not sure). It looks something like this:

![image](https://user-images.githubusercontent.com/6682655/58200978-a775f980-7cf1-11e9-8a57-9b26825270a2.png)

**Comment by fabiospampinato** [comment]
Yeah I use `fzf` (and the lovely Hyper!) too. Something like that could be implemented I guess, maybe using `fzf` directly could be problematic as we would have to ship it with the app across platforms, which is probably not trivial, and it will return different results than those provided by the app, while I think ideally the two should be the same.

**Comment by dag-err** [comment]
> An official CLI app would be useful.

Hi there, my very first comment here (so sorry in advance if I didn't do enough homework and looked for all the issues).

I think, ATM, a lot could be done already in the CLI realm (aside from having a specific client or not) by having the possibility to open a specific file as a `notable` CLI option.

Something like:

`notable -o title`

that would open notable if not running or if running it will show the window and move to that file.

Would that be a possibility?

**Comment by fabiospampinato** [comment]
> Would that be a possibility?

Why not, yes.

**Comment by dag-err** [comment]
Do you want me to open a new enhancement request for it or shall it be considered as part of the CLI app functionalities?

**Comment by fabiospampinato** [comment]
@dag-err let's keep a single issue for the CLI app, at least until a CLI gets implemented.

**Comment by fabiospampinato** [comment]
There isn't a CLI app yet, but we now have a custom "notable://" protocol, which should be able to replace some aspects of a CLI: https://www.reddit.com/r/Notable/comments/fcw7ii/preview_command_palette_macro_command_links_to/

**Comment by k15r** [comment]
> Yes, perhaps `fzf` can be used directly (not sure). It looks something like this:
> 
> ![image](https://user-images.githubusercontent.com/6682655/58200978-a775f980-7cf1-11e9-8a57-9b26825270a2.png)
> 
> Also, I can search through my notes in the `notes` directory. The CLI could probably make this simpler and open notable with the selected note, just throwing ideas 😄
> 
> ![image](https://user-images.githubusercontent.com/6682655/58201277-4f8bc280-7cf2-11e9-8889-b834ffcfcd0f.png)

```
_note_search () {
	RG_PREFIX='rg --files-with-matches --no-messages --ignore-case'
	rg --files-with-matches --no-messages '' ${NOTES_DIR} | gxargs -d '\n' -n1 basename | fzf --bind "change:reload:$RG_PREFIX {q} ${NOTES_DIR} | gxargs -d '\n' -n1 basename || true " --preview "highlight -O ansi ${NOTES_DIR}/{} 2> /dev/null | rg --colors 'match:bg:yellow' --ignore-case --pretty --no-line-number --context 10 {q} || rg --ignore-case --pretty --no-line-number --context 10 {q} ${NOTES_DIR}/{}" --preview-window 'right:70%' --phony
}
```

its not beautiful, but does the job:

**Comment by RyanGreenup** [comment]
I love using `rg` and `--preview`, but if you use [skim](https://github.com/lotabout/skim) it can live update the ripgrep search interactively and then switch to searching by name, much better than using something like `C-S-f` in *VS Code* for instance.

I found the search lacking though because it's only grep not a search engine, so I ended up 
playing with [recoll](https://www.lesbonscomptes.com/recoll/) and made [this simple simple shell function](https://github.com/RyanGreenup/Note-Taking-Tools/blob/master/Terminal-Skim-Recoll/Terminal-Skim-Recoll.md), recoll works similarly but implements some type of term-frequency weighting algorithm to match results across the whole document, so it looks like this:


<img src=https://user-images.githubusercontent.com/47832594/85483021-9c0b9680-b5b3-11ea-9d90-1cfee383be6f.png alt="alt text" width=200>


<img src=https://user-images.githubusercontent.com/47832594/85483084-bba2bf00-b5b3-11ea-9159-003fb13e40df.png alt="alt text" width=400>

well worth setting up if you work inside *Vim* / *Emacs* / *VSCode* and want to access your notes

**Comment by nsheff** [comment]
Here's another idea for the CLI -- given a note title (or some other way of identifying a note, like a `notable://...` URI), give me back the content of the note.

SO, something like:

```
notable --retrieve notable://note/my%20title
```
